### ISSUE TITLE
It should be more obvious to admins how to create users

### ISSUE THREAD

**Issue by myieye** [sillsdev/languageforge-lexbox#666]
We initially tried to avoid letting admin create users.
We've realized/decided now that there is a good use case for that.
- Users aren't tech savvy
- Users don't speak English (the language of our UI)
- etc.
I've directed a small handfull of admins to /register for creating admins.
We're about to release our Bulk Add/Create users tool.

What's REALLY missing now is UI that gets an admin from the dashboard to these 2 ways to create users. Currently, if an admin wants to create a user they still need to ask.

**Comment by myieye** [comment]
@hahn-kev we discussed this today. What do you think about this:

I like the idea of fixing this in tandem with improving the admin workflow for creating users:

- Add a create user button on the admin dashboard which
- Opens a modal (?) with the "Register user" form in it (can the admin enter a login/username instead of an email address?)
- Somewhere on the form (at the top?) is a note/alert: "Did you know that you can bulk create users with the "Bulk Add/Create users" button when viewing a project?"
- Users created this way have their "CreatedBy" column populated

**Comment by myieye** [comment]
### Discussion from our agend doc

@hahn-kev 
> I'm not sure... if users can be invited by email, then they can probably make their own account from that invite. Otherwise they should probably be guest users without emails in which case use the bulk add and the admin can specify their password. I'm not sure I want yet another way for a user account to get created (register, invite and bulk add)

@myieye 

> I don't quite understand tihs: "if users can be invited by email, then they can probably make their own account from that invite"
> Are you suggesting that users who can use email are likely tech-savvy enough to register themselves?
> 
> 
> Here are the three things that I think are important to us:
> - Admins need a "Create user" button on the dashboard (wether it gives them a form or tells them to use bulk create) so that > there's a visible path for creating users
> - Admins shouldn't need to use the register page to create users (it's a workaround. They get logged in as the user they're > registering and we don't record that that user was created by an admin)
> - We prefer email addresses over logins
> 
> 
> So:
> If we upgrade/fix the bulk create so that it works with email addresses. And we explain/make that feature visible on the admin dashboard, then that's likely sufficient. #691

@hahn-kev 
> well right now if you add a member via email and they don't have an account (not bulk add) they will be sent an invite to create an account and they will be added to that project automatically. That seems like the majority use case for new users no? Either way I'd rather go through the invite process for users with an email. They've already said they find it hard to type in the email address of a user to invite them to a project, it doesn't make sense to then also ask them to set a password and their name etc.
> 
> I agree bulk add should work with email addresses too, I didn't realize it was blocked from the front end.

@hahn-kev 
> to your question, yes I think if they have an email we can expect them to register on their own. If they can't register on their own maybe they should just have a guest user (or whatever we're calling users without emails)

### TLDR;

It doesn't sound like we want to extract the register page.

#### Use case 1: An admin wants to add a user that would have trouble using our UI

👉 The admin should use the bulk add tool.

If the user can't register then they presumably won't generally login and probably won't need to manage projects.

#### Use case 2: An admin wants to add a user that could use our UI

👉 The admin should either invite the user via email or just direct them to the register page

**Comment by rmunn** [comment]
I'm not sure that our decision was the right one; I think extracting the register page might actually be a good idea after all, and that's the approach I ended up taking in #736. I'll pause work on this and discuss it further at the meeting tomorrow.

**Comment by rmunn** [comment]
Discussion from our meeting on Friday produced the following summary:

> Chris says that as a Language Depot admin / tech support role, he’d prefer to be able to do the entire workflow start to finish: create the user account and assign it to the project, and not have to wait for the user to check his email to accept the email invitation.

I tend to agree. We want users to be able to register because we're trying to remove some of the burden on admins of the existing workflow where the admin has to do everything. However, there might be times when a user is having email problems, and wants an admin to help him create his account and a project. I don't think we should force admins to go through the bulk-create process and its poor UI in that situation; I'd much prefer to have a "Create User" button that does what the admin would expect it to do.

**Comment by hahn-kev** [comment]
My initial thoughts right now is that maybe we surface a way for admins to create guest users outside of the bulk add feature, we would allow them to set a name for the user, and even an email (though it would still need to be verified by the user) if they wanted. This is pending a chat with Tim.

**Comment by rmunn** [comment]
The scenario that I'm thinking of here is *not* guest users, but users who will be given project-manager rights.

To expand a little on what I mean by "poor UI in that situation": you're a Language Depot / LexBox admin who's been asked by a user "I'm having trouble accessing my email so I can't click on project-invite emails. Could you help me create my language project so I can do Send/Receive from FieldWorks?". (And you know this isn't a case of phishing). Would you rather have an interface that looks like this to create the user?

![image](https://github.com/sillsdev/languageforge-lexbox/assets/90762/556bc017-d59c-4a38-b833-0855a12fd982)

Or one that looks like this?

![image](https://github.com/sillsdev/languageforge-lexbox/assets/90762/92c6e5cf-41f0-4f56-9ae5-f1bab22f69c6)

The latter currently allows creating accounts with email addresses, but not display names. (The backend would parse `"Robin Munn" <rmunn@example.com>` correctly and set the display name, but the frontend currently disallows that syntax due to an overly-restrictive validation regex, so there's no way for an admin to set the display name of the user in a single step via that tool). And the "shared password" field looks ominous when creating a single account. (And, indeed, it sets the password strength to 0, for good reasons, so that the newly-created user will actually be forbidden from being a manager once we implement the second half of #57. This fact is not currently communicated to the user or the admin.)

Yes, after the account is created, the user could log in and change his display name, change his password, and so on. But all of that is extra work we'd be requiring the admins to do (and communicate to the users), rather than a one-step-and-done process.

I agree that the ideal is to have the users do the work of registering themselves, etc. But in those cases where someone comes to one of our tech-support people who has admin rights on LexBox, we should give the admins the tools to solve the problem right then and there, while the user is standing there by the admin's desk.

Basically, when there's a workflow that people are used to using and we want to guide them to a different one, **I don't want to say "No, but". I want to say "Yes, and".**

I don't want to say "**No,** you may not have a "Create User" button that simply creates a user. **But** you should go to the project page instead, so here's a button that doesn't actually create users, it just tells you what the new process is." Instead, I want to say "**Yes,** you may create users if you want to. **And** did you know that there's an easier way? Just go to the project page and you can invite them to the project *and* register them all at once."

**Comment by myieye** [comment]
@hahn-kev Can you explain what you mean exactly by "guest users". I feel like you've used that term quite a bit and it's not totally clear to me.

**Comment by hahn-kev** [comment]
Users created via the bulk add are guest users, not sure what else to call them.

**Comment by rmunn** [comment]
This is turning into [a much larger task](https://github.com/sillsdev/languageforge-lexbox/pull/736#pullrequestreview-2055873940) than the original design called for. I'm going to bump the estimate all the way from Small to Large, because of how much the scope of this feature has grown in the code-review process.
